README.md

```diff
@@ -1,30 +1,135 @@
+
+The plugin can also modify the source file with the updated CSS -- with correct fallback values. See `write` under [options]
+
+> If you use a code formatter like [prettier], you may need to re-run it after the plugin modifies the source file.
+
+## Options
+
+The plugin accepts 3 options:
+
+**`properties`**
+
+An object containg key and fallback values of custom properties
+
+| type   | required | default |
+| ------ | -------- | ------- |
+| Object | true     | N/A     |
+
+**`write`**
+
+Writes the source CSS file with the updated CSS
+
+| type    | required | default |
+| ------- | -------- | ------- |
+| boolean | false    | false   |
+
+**`callback`**
+
+Callback to handle the response of the plugin. The callback accepts 2 arguments:
+
+- error: an error object
+- data: the response
+
+| type     | required | default  |
+| -------- | -------- | -------- |
+| Function | false    | () => {} |
+
+## Response
+
+The response is a list of object of custom properties with incorrect fallback values. The plugin's result also contains the response -- `result.wrongProps`. Shape of the object:
+
+- `path` {string} - absolute path of the CSS file
+- `line` {number} - line number containing the custom property
+- `key` {string} - key of the custom property
+- `current` {string} - current fallback value
+- `expexted` {string} - expected fallback value
+
+## Example
+
+Sample `properties` option:
+
+```json
+{
+  "--border-radius": "4px",
+  "--color-red": "#ff0000",
+  "--danger-border":
+    "var(--border-radius, 4px) solid var(--color-red, #ff0000)",
+};
+```
+
+A sample CSS:
 
 ```css
 .foo {
-    /* Input example */
+  border: var(
+    --danger-border,
+    var(--border-radius, 5px) solid var(--color-red, #fa0000)
+  );
+}
+
+.bar {
+  --danger-text: var(--color-red, red);
+  color: var(--danger-text);
 }
 ```
 
-```css
+The resultant CSS that the plugin produces:
+
+```diff
 .foo {
-  /* Output example */
+  border: var(
+    --danger-border,
+-    var(--border-radius, 5px) solid var(--color-red, #fa0000)
++    var(--border-radius, 4px) solid var(--color-red, #ff0000)
+  );
 }
+
+.bar {
+-  --danger-text: var(--color-red, red);
++  --danger-text: var(--color-red, #ff0000);
+  color: var(--danger-text);
+}
+```
+
+List of the incorrect custom properties ([response]).
+
+```js
+[
+  {
+    path: undefined,
+    line: 2,
+    key: "--danger-border",
+    current: "var(--border-radius, 5px) solid var(--color-red, #fa0000)",
+    expected: "var(--border-radius, 4px) solid var(--color-red, #ff0000)",
+  },
+  {
+    path: undefined,
+    line: 9,
+    key: "--color-red",
+    current: "red",
+    expected: "#ff0000",
+  },
+];
 ```
 
 ## Usage
 
+### Add to plugins list
+
 **Step 1:** Install plugin:
 
 ```sh
 npm install --save-dev postcss postcss-customprop-validate
 ```
 
-**Step 2:** Check you project for existed PostCSS config: `postcss.config.js`
+**Step 2:** Check you project for existing PostCSS config: `postcss.config.js`
```

Comment:
typo. should be `your`.

Reply:
Ahh.. came with the boilerplate. Did not check it!